### ISSUE TITLE
Numeric difference in logits during decoding

### ISSUE THREAD

**Issue by njellinas** [karpathy/nanoGPT#396]
I ran the simplest example for shakespeare, it doesn't matter if the model is trained or not I just obtained the first checkpoint at 10 steps. Then while doing inference I saved the sequence of the logits in each step and noticed that just after the 2nd step, there is a numerical difference between the saved logits of the previous step and the logits on the output of the transformer. E.g. the first logit should be exactly the same when running for the second token, but if you calculate the difference between them there is a difference of like 10e-4.
I ran this example here because I noticed the same thing in one of my models which has continuous outputs. In the first iteration the first predicted output is different. In the second iteration and all following it remains different, BUT the second output is exactly the same (torch.equal). AND in the final 3 iterations (I ran for 19) the second output becomes different again.
